paf-mvp-cmp/src/controller.ts

```diff
@@ -0,0 +1,565 @@
+          [Marketing.custom, this.config.replace(this.locale.snackbarHeadingCustomized)],
+          [Marketing.notSet, this.config.replace(this.locale.snackbarHeadingCustomized)],
+        ])
+      )
+    );
+    this.model.pref.addBinding(
+      new BindingElement<PreferencesData, Model>(
+        this.view,
+        'ok-ui-snackbar-body',
+        new Map<PreferencesData, string>([
+          [Marketing.personalized, this.config.replace(this.locale.snackbarBodyPersonalized)],
+          [Marketing.standard, this.config.replace(this.locale.snackbarBodyStandard)],
+          [Marketing.custom, this.config.replace(this.locale.snackbarBodyCustomized)],
+          [Marketing.notSet, this.config.replace(this.locale.snackbarBodyCustomized)],
+        ])
+      )
+    );
+    this.model.pref.addBinding(new BindingShowRandomId(this.view, 'ok-ui-settings-rid', this.model));
+    this.model.onlyThisSite.addBinding(new BindingChecked(this.view, 'ok-ui-only-this-site'));
+    this.model.onlyThisSite.addBinding(
+      new BindingThisSiteOnly(this.view, 'ok-ui-only-this-site-container', this.config)
+    );
+    for (let id = Model.MinId; id <= Model.MaxId; id++) {
+      this.model.tcf.get(id).addBinding(new BindingChecked(this.view, `ok-ui-preference-${id}`));
+    }
+    this.model.all.addBinding(new BindingChecked(this.view, 'ok-ui-preference-all'));
+    this.model.canSave.addBinding(new BindingButton(this.view, 'ok-ui-button-save'));
+    this.model.rid.addBinding(new BindingDisplayRandomId(this.view, 'ok-ui-display-rid'));
+  }
+
+  /**
+   * Loads the data in the following order.
+   * 1. Local storage
+   * 2. Global storage without using a redirect
+   * If config.displayIntro is false then;
+   * 3. Global storage using a redirect.
+   * If config.displayIntro is true then the intro card is displayed the redirect will only occur if the user selects
+   * proceed.
+   */
+  private async load() {
+    if (this.getIdsAndPreferencesFromLocal()) {
+      return;
+    }
+    if (await this.getIdsAndPreferencesFromGlobal(false)) {
+      return;
+    }
+    if (this.config.displayIntro === false) {
+      await this.getIdsAndPreferencesFromGlobal(true);
+    }
+    return;
+  }
+
+  /**
+   * Gets the Ids and preferences which might involve a redirect completing this instance if the redirect is allowed.
+   * If data is returned then the model is updated and the display method called.
+   * @param triggerRedirectIfNeeded: `true` if redirect can be triggered immediately, `false` if it should wait
+   * @returns true if the data is valid, otherwise false
+   */
+  private async getIdsAndPreferencesFromGlobal(triggerRedirectIfNeeded: boolean) {
+    // TODO: These values are used by the refresh method. Remove them to force processing a redirect if needed and
+    // allowed.
+    removeCookie(Cookies.identifiers);
+    removeCookie(Cookies.preferences);
+    removeCookie(Cookies.lastRefresh);
```

Comment:
If understand this method correctly, the idea is to get data from global, ignoring local values if they exist (ensure "fresh" data).
If this is the case, I have a couple of concerns:
1. only `lastRefresh` would need to be deleted. Its purpose is to keep the local values, to be able to compare when fresh data comes from the operator. Also, I think it's a better approach to only delete the `lastRefresh` cookie so that if the call to the operator fails for any reason, we haven't removed the local ids and prefs cookies
2. it would make more sense to handle this inside PAF-lib. As a separate method, or even as a parameter to refreshIdsAndPreferences
3. if the logic is that you want to make sure to get fresh values before showing the UI (which I think would be a good idea), then we should do the same for the other UI. This is not the case today (we simply get current 1PC values). We need more consistency.

Reply:
@jwrosewell would you please confirm my interpretation was correct?
The objective was to make sure we always get fresh data from the operator before showing the UI?
Thanks.

Reply:
Reading the code again, I think my interpretation is not exactly accurate.
I thought this "force refresh" was done when a user tries to EDIT existing preferences, but in fact I think it is only triggered after the first "agree" click, so I have the feeling it is a little redundant, because this will only happen on a first visit, wouldn't it?
Anyway, I think there is room for a new "resetLocalCookies" method and will plan to add it to PAF-lib.